virt-v2v/cold/entrypoint

```diff
@@ -1,23 +1,66 @@
 #!/usr/bin/env bash
+shopt -s nullglob
 
-echo "Run virt-v2v with the following input:"
-cat /mnt/v2v/input.xml
-
-virt-v2v -v -x -i libvirtxml -o null --debug-overlays --no-copy --root=first /mnt/v2v/input.xml
-[ $? != 0 ] && exit 1
-
-echo "Conversion successful. Committing all overlays to local disks."
-for OVERLAY in /var/tmp/*.qcow2
-do
-	if ! qemu-img commit -p "$OVERLAY"
-	then
-		echo Failed to commit overlay "$OVERLAY"!
-		echo Unable to complete import!
-		exit 1
-	fi
+echo "Preparing virt-v2v"
+
+# This variable is used to build the list of arguments for virt-v2v.
+args=()
+
+# Temporary storage location
+DIR="/tmp/v2v"
+mkdir -p $DIR
+args=("${args[@]}"
+    -o local
+    -os "$DIR"
+)
+
+# Generate disk name suffix from disk number. E.g. "c" (as in "sdc") for
+# 3rd disk.
+gen_name() {
+    local chars=({a..z})
+    local cnt="${#chars[@]}"
+    local i=$(($1 % cnt))
+    local rest=$(($1 / cnt))
+    local prefix=""
+    if [ $rest -gt 0 ] ; then
+        prefix="$(gen_name $rest)"
+    fi
+    echo "$prefix${chars[$i-1]}"
+}
+
+# Disks on filesystem storage.
+# e.g.: /mnt/disks/disk0/disk.img -> vmName-sda
+for disk in /mnt/disks/disk[0-9]* ; do
+	num="${disk:15}"
+	ln -s "$disk/disk.img" "$DIR/$V2V_vmName-sd$(gen_name "$((num+1))")"
+done
+# Disks on block storage.
+# e.g.: /dev/block0 -> vmName-sda
+for disk in /dev/block[0-9] ; do
```

Comment:
I don't think we'll have more than 10 volumes but shouldn't it be /dev/block[0-9]* as above?

Reply:
It definitely should!